### PR TITLE
add `D.Gumbel`

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**

### CODE REVIEW

File: tests/chainer_tests/distributions_tests/test_gumbel.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+from chainer import distributions
+from chainer import testing
+from chainer import utils
+import numpy
```
Comment (toslunar) [inline tests/chainer_tests/distributions_tests/test_gumbel.py]: Could you fix order of the imports?

File: chainer/distributions/gumbel.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    def scale(self):
+        return self.__scale
+
+    @property
+    def batch_shape(self):
+        return self.loc.shape
+
+    @property
+    def entropy(self):
+        return exponential.log(self.scale) + EULER + 1.
```
Comment (toslunar) [inline chainer/distributions/gumbel.py]: `... + (EULER + 1)` would be more efficient

**Comment by toslunar** [comment]
Jenkins, test this please.

File: tests/chainer_tests/distributions_tests/test_kldivergence.py [hunk 3]
```diff
@@ -222,6 +229,17 @@ def test_gamma_gamma_gpu(self):
         dist2 = self.make_gamma_dist(True)
         self.check_kl(dist1, dist2)
 
+    def test_gumbel_gumbel_cpu(self):
```
Comment (toslunar) [inline tests/chainer_tests/distributions_tests/test_kldivergence.py]:
```suggestion
@testing.with_requires('scipy')
    def test_gumbel_gumbel_cpu(self):
```

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

**Pull request merged**